### ISSUE TITLE
Add/Remove Columns

### ISSUE THREAD

**Issue by madhan-dhamodaran** [adazzle/react-data-grid#1240]
- Have you already searched for similar issues? Please help us out and double-check first!

- Also, please don't be that person who deletes this template. It's here for a reason.

- Thanks!

---

### Which version of React JS are you using?

✅ Officially supported ✅
- [ ] v15.4.x

⚠️ Not officially supported, expect warnings ⚠️
- [ ] v15.5.x
- [ ] v15.6.x

☣️ Not officially supported, expect warnings and errors ☣️
- [ ] v16.x.x

---

### Which browser are you using?

✅ Officially supported ✅
- [ ] IE 9 / IE 10 / IE 11
- [ ] Edge
- [ ] Chrome

⚠️ Not officially supported, but "should work" ⚠️
- [ ] Firefox
- [ ] Safari

---

### I'm submitting a ...

- [ ] 🐛 Bug Report
- [ ] 💡 Feature Request

> 👋 Need general support? Not sure about how to use React itself, or how to get started with the Grid?
> Please do not submit support request here. Instead see

> https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md

---

### Issue Details

```
Please include:
- What the current behavior is
- What the desired behvaior is
- (If Bug) Steps to reproduce the issue
- (If Feature) The motivation / use case for the feature

We especially love screenshots / videos of problems, and remember
The Best Issue Is A Pull Request™
```
@amanmahajan7 @rozeskjm @azu 

Hi All,

This is a feature proposal. 

I would like to add a feature to provide the user the ability to customize the columns that they want to display. 

Basically, if they have to show around 50 columns, they can decide to show just 5 and the rest would be unselected and displayed in an adjacent section. 

They can go ahead and choose to add that column.

Thanks,
Madhan

**Comment by madhan-dhamodaran** [comment]
@amanmahajan7 @rozeskjm @azu, 

I have already added this ability and I will be raising a PR by this weekend. Sorry I was in a tight schedule so I couldn't raise an issue earlier. 

But, this is ready to go now. 

Thanks,
Madhan

**Comment by madhan-dhamodaran** [comment]
All, 

I am still working on this. I will raise a PR ASAP

Thanks,
Madhan

**Comment by madhan-dhamodaran** [comment]
All,

I am still working on this. I will raise a PR ASAP

Thanks,
Madhan

**Comment by madhan-dhamodaran** [comment]
All,

I am still working on this. I will raise a PR ASAP

Thanks,
Madhan

**Comment by madhan-dhamodaran** [comment]
All,

I am still working on this. I will raise a PR ASAP

Thanks,
Madhan
